packages/test-app/tests/integration/components/fade-text-test.js

```diff
@@ -0,0 +1,155 @@
+import { module, test, skip } from 'qunit';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { component } from 'ilios-common/page-objects/components/fade-text';
+
+module('Integration | Component | fade-text', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders empty', async function (assert) {
+    await render(hbs`<FadeText />`);
+    assert.strictEqual(component.text, '');
+
+    await render(hbs`<FadeText></FadeText>`);
+    assert.dom(this.element).hasText('');
+  });
+
+  test('it renders short text in full', async function (assert) {
+    const shortText = 'template block text';
+    this.set('shortText', shortText);
+
+    await render(hbs`<FadeText @text={{this.shortText}} />
+`);
+    assert.strictEqual(component.text, shortText);
+    assert.notOk(component.collapse.isVisible);
+    assert.notOk(component.expand.isVisible);
+
+    await render(hbs`
+      <FadeText>
+        {{this.shortText}}
+      </FadeText>
+    `);
+    assert.dom(this.element).hasText(shortText);
+  });
+
+  module('fading (flaky)', () => {
+    skip('it fades tall text', async function (assert) {
+      const longHtml = `
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.<br />
+        An objective description so long that it fades.
+      `;
+      const fadedClass = 'is-faded';
+      this.set('longHtml', longHtml);
+
+      await render(hbs`<FadeText @text={{this.longHtml}} />`);
+
+      assert.dom('.display-text-wrapper', this.element).hasClass(fadedClass);
+
+      await component.expand.click();
+
+      assert.dom('.display-text-wrapper', this.element).doesNotHaveClass(fadedClass);
+      await component.collapse.click();
+      assert.dom('.display-text-wrapper', this.element).hasClass(fadedClass);
+
+      await render(hbs`
+        <FadeText @text={{this.longHtml}} as |displayText expand collapse getTextDims updateTextDims isFaded expanded|>
+          <div class="display-text-wrapper{{if isFaded ' is-faded'}}">
```

Comment:
Big fan of testing the block invocation here, but the actual content of the test probably shouldn't mirror the component without the block. Instead just enough HTML should go here to ensure that the passed properties are as you would expect. I think the amount of content needed in the block invocation may be an issue with the existing truncation component as well. I think it's ok right now, but might be worth another pass in the future. I wonder if we could split out different parts of this into [named blocks](https://guides.emberjs.com/release/components/block-content/) so they don't have to be overridden.